backlinks/obs_filter.js

```diff
@@ -0,0 +1,80 @@
+var nest = require('depnest')
+var pull = require('pull-stream')
+var MutantPullReduce = require('mutant-pull-reduce')
+var Value = require('mutant/value')
+
+exports.needs = nest({
+  'sbot.pull.backlinks': 'first'
+})
+
+exports.gives = nest("backlinks.filter.obs", true)
+
+/**
+ * sbot.filter.obs returns an observable list of messages that link
+ * back to the message with the given message ID (@id). Only messages that
+ * pass the filter function are added to the list.
+ *
+ * When a message arrives, if passing it to the filter function (@filterFn)
+ * does not result in it returning 'true' the message is not added to the
+ * observable list.
+ *
+ * A 'sync' observable property is also added to the returned observable
+ * which is 'true' when all previously seen messages are caught up with.
+ *
+ * Note: Unlike backlinks.obs.for this does not cache the observable for
+ * callers that supply the same arguments.
+ */
+exports.create = function(api) {
+  function pullFilterReduceObs(id, filterFn) {
+    if (!id || typeof(id) !== "string") {
+      throw new Error("id must be a string.")
+    }
+
+    if (!filterFn || typeof(filterFn) !== "function") {
+      throw new Error("filterFn must be a function.")
+    }
+
+    var sync = Value(false)
+
+    var msgBacklinks = api.sbot.pull.backlinks({
+      query: [{
+        $filter: {
```

Comment:
I see that you are passing in a function as a filter. But it is also interesting to note that the query here allows you to pass in JSON properties to filter. The advantage to the JSON filter instead of a function is that the filtering will happen in the sbot process which will make it a little bit more faster and efficient (since it doesn't have to send all the extra data down the wire).

The filter method makes sense, but an idea for improvement and making this module more versatile (maybe later) would be to also accept an object as a filter, and then inject it into the query filter.
 
It is using this: https://github.com/dominictarr/map-filter-reduce